tests/src/test/scala/ch/epfl/directembedding/test/TypeOverridingSpec.scala

```diff
@@ -0,0 +1,28 @@
+package ch.epfl.directembedding.test
+
+import ch.epfl.directembedding.test.example._
+import org.scalatest.{ FlatSpec, ShouldMatchers }
+
+class TypeOverridingSpec extends FlatSpec with ShouldMatchers with ExampleTester {
+
+  "dsl" should "allow custom overriding of ints" in {
+    runTest(
+      dsl {
+        val i = 1
+        i + 1
+      }) should be(IntPlus(Infix_ValDef(1), 1))
+  }
+
+  // This case is different from IntPlus because the
+  // type of `s` is String and the result is of type Int
+  "dsl" should "allow custom overriding of strings" in {
+    runTest(
+      dsl {
+        val s = "foobar"
+        s.length
+      }) should be(StringLength(Infix_ValDef("foobar")))
+  }
+
+  // TODO: Add test cases for all branches in ReifyAsEmbedding
```

Comment:
Remove? There are a lot of untested cases but I thought it might be good to do that through another PL.